src/rest/resource.cpp

```diff
@@ -800,6 +803,33 @@ void Resource::DatasetPending(const Request &aRequest, Response &aResponse) cons
     Dataset(DatasetType::kPending, aRequest, aResponse);
 }
 
+void Resource::GetCoprocessorVersion(Response &aResponse) const
+{
+    std::string coprocessorVersion;
+    std::string errorCode;
+
+    coprocessorVersion = mHost->GetCoprocessorVersion();
+    coprocessorVersion = Json::String2JsonString(coprocessorVersion);
+
+    aResponse.SetBody(coprocessorVersion);
+    errorCode = GetHttpStatus(HttpStatusCode::kStatusOk);
+    aResponse.SetResponsCode(errorCode);
+}
+
+void Resource::CoprocessorVersion(const Request &aRequest, Response &aResponse) const
+{
+    std::string errorCode;
+
+    if (aRequest.GetMethod() == HttpMethod::kGet)
+    {
+        GetCoprocessorVersion(aResponse);
+    }
+    else
+    {
+        ErrorHandler(aResponse, HttpStatusCode::kStatusMethodNotAllowed);
```

Comment:
Perhaps we can have a refactor to these resource handlers to return whether the request is successfully processed so that this error can be responded in one place.

Reply:
Do you mean in this specific PR or in general?